### ISSUE TITLE
Gnome shell sometimes crashes when an application gets launched

### ISSUE THREAD

**Issue by chbrueck** [home-sweet-gnome/dash-to-panel#1712]
**Describe the bug**
This extension sometimes crashes gnome shell when an application gets launched. This crash is extremely rare and happens when the extension tries to add a taskbar icon for the starting application. I sadly cannot reproduce it consistently but I managed to grab the logs after it happened.
Here are the logs from the systemd journal:

```
// This happened shortly before starting the application but it still seems relevant
Jul 13 18:23:18 gnome-shell[1823]: JS ERROR: TypeError: this._raisedClone.source is null
                                    onComplete@/home/christian/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1499:21
                                    _makeEaseCallback/<@resource:///org/gnome/shell/ui/environment.js:153:13
                                    _easeActor/<@resource:///org/gnome/shell/ui/environment.js:240:64

// Started the application
Jul 13 18:25:43 gnome-shell[1823]: Object .Gjs_dash-to-panel_jderose9_github_com_taskbar_TaskbarItemContainer (0x55d1e9f21630), has been already disposed — impossible to access it. This might be caused by the object having been destroyed>
Jul 13 18:25:43 gnome-shell[1823]: == Stack trace for context 0x55d1e97b3df0 ==
Jul 13 18:25:43 gnome-shell[1823]: #0   55d1efa6b8e8 i   /home/christian/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1398 (3f0dd997330 @ 27)
Jul 13 18:25:43 gnome-shell[1823]: #1   55d1efa6b860 i   /home/christian/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1433 (3f0dd997420 @ 22)
Jul 13 18:25:43 gnome-shell[1823]: #2   55d1efa6b748 i   /home/christian/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:148 (3f0dd995290 @ 1097)
Jul 13 18:25:43 gnome-shell[1823]: #3   7fff329e8880 b   /home/christian/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:874 (3f0dd982c90 @ 132)
Jul 13 18:25:43 gnome-shell[1823]: #4   7fff329e88f0 I   self-hosted:202 (12762808b1a0 @ 272)
Jul 13 18:25:43 gnome-shell[1823]: #5   7fff329e89f0 b   /home/christian/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:870 (3f0dd982c40 @ 1069)
Jul 13 18:25:43 gnome-shell[1823]: #6   7fff329e8b10 b   /home/christian/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:916 (3f0dd982b50 @ 965)
Jul 13 18:25:43 gnome-shell[2199]: (EE) failed to read Wayland events: Broken pipe
Jul 13 18:25:43 systemd[1616]: org.gnome.Shell@wayland.service: Main process exited, code=killed, status=11/SEGV
Jul 13 18:25:43 systemd[1616]: org.gnome.Shell@wayland.service: Failed with result 'signal'.
Jul 13 18:25:43 systemd[1616]: org.gnome.Shell@wayland.service: Triggering OnFailure= dependencies.
Jul 13 18:25:43 systemd[1616]: org.gnome.Shell@wayland.service: Consumed 11min 24.530s CPU time.
```

**Linux distribution and version**
Debian GNU/Linux bookworm/sid

**GNOME Shell version**
GNOME Shell 42.2

**Dash-to-Panel version**
Version 50

**Where was Dash-to-Panel installed from?**
It was installed from extensions.gnome.org

**Comment by williamd5** [comment]
Could be related to my issue #1664

**Comment by Spurlos** [comment]
Can confirm similar behaviour on app launch.

stack:
```
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: Object .Gjs_dash-to-panel_jderose9_github_com_taskbar_TaskbarItemContainer (0x55dd036445c0), has been already disposed — impossible to access it. Thi>
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: == Stack trace for context 0x55dcfd46b170 ==
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #0   7ffd5c4a2910 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1398 (10e7ff1ccdd0 @ 27)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #1   55dcfd845018 i   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1433 (10e7ff1ccec0 @ 22)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #2   7ffd5c4a44e0 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:148 (10e7ff1c5d30 @ 1055)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #3   7ffd5c4a5750 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:880 (10e7ff1ba740 @ 132)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #4   7ffd5c4a5860 b   self-hosted:164 (21f22348c330 @ 272)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #5   7ffd5c4a5960 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:876 (10e7ff1ba6f0 @ 1068)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #6   7ffd5c4a5a80 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:922 (10e7ff1ba600 @ 962)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #7   7ffd5c4a7100 b   resource:///org/gnome/shell/ui/layout.js:1029 (21f2234cc3d0 @ 422)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #8   7ffd5c4a71d0 b   self-hosted:1121 (21f22347eec0 @ 407)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: GNOME Shell crashed with signal 11
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: == Stack trace for context 0x55dcfd46b170 ==
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #0   7ffd5c4a2910 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1401 (10e7ff1ccdd0 @ 288)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #1   55dcfd845018 i   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1433 (10e7ff1ccec0 @ 22)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #2   7ffd5c4a44e0 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:148 (10e7ff1c5d30 @ 1055)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #3   7ffd5c4a5750 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:880 (10e7ff1ba740 @ 132)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #4   7ffd5c4a5860 b   self-hosted:164 (21f22348c330 @ 272)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #5   7ffd5c4a5960 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:876 (10e7ff1ba6f0 @ 1068)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #6   7ffd5c4a5a80 b   /home/spurlos/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:922 (10e7ff1ba600 @ 962)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #7   7ffd5c4a7100 b   resource:///org/gnome/shell/ui/layout.js:1029 (21f2234cc3d0 @ 422)
Dec 12 14:29:29 spurlos-ThinkPad-E580 gnome-shell[2329752]: #8   7ffd5c4a71d0 b   self-hosted:1121 (21f22347eec0 @ 407)
```

**Comment by Spurlos** [comment]
according to stack trace extension is:
- trying to update the clone position while adding icon to taskbar
```
1431-1433
        // Update clone position when an item is added to / removed from the taskbar.
        let taskbarBox = this.get_parent();
        let taskbarBoxAllocationChangedId = taskbarBox.connect('notify::allocation', () => 	this._updateCloneContainerPosition(cloneContainer));
```
- fails to access the TaskbarItemContainer object as it was disposed for some reason
```
1397-1404
    _updateCloneContainerPosition(cloneContainer) {
        let [stageX, stageY] = this.get_transformed_position();

        cloneContainer.set_position(
            stageX - this._dtpPanel.panelBox.translation_x - this.translation_x,
            stageY - this._dtpPanel.panelBox.translation_y - this.translation_y
        );
    }
```

Hope this would help to solve or at least mitigate the issue, as it causes DE crash numerous times during regular usage (like on each other launch of any app )

**Comment by charlesg99** [comment]
There was a fix for a crash seen in #1666, let me know if that fixes this issue as well

**Comment by Loatchi** [comment]
I'm having the same problem:
- gnome-shell-44.1
- dash-to-panel-56 (from extensions.gnome.org)
- Gentoo Linux (Gentoo Base System release 2.13)

The logs:
```
May 08 18:05:44 acer gnome-shell[2599]: Object .Gjs_dash-to-panel_jderose9_github_com_taskbar_TaskbarItemContainer (0x563b81ee2fa0), has been already disposed — impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
May 08 18:05:44 acer gnome-shell[2599]: == Stack trace for context 0x563b7d9d28e0 ==
May 08 18:05:44 acer gnome-shell[2599]: #0   7ffd1ec07930 b   /home/francois/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1396 (335781993790 @ 27)
May 08 18:05:44 acer gnome-shell[2599]: #1   563b80e04ce8 i   /home/francois/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:1431 (335781993880 @ 22)
May 08 18:05:44 acer gnome-shell[2599]: #2   563b80e04bd0 i   /home/francois/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js:148 (3357819806f0 @ 1055)
May 08 18:05:44 acer gnome-shell[2599]: #3   563b80e04b30 i   /home/francois/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:820 (12972fa08b00 @ 132)
May 08 18:05:44 acer gnome-shell[2599]: #4   7ffd1ec0b850 b   self-hosted:164 (36f0dc491ab0 @ 272)
May 08 18:05:44 acer gnome-shell[2599]: #5   563b80e04a50 i   /home/francois/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:816 (12972fa08ab0 @ 1068)
May 08 18:05:44 acer gnome-shell[2599]: #6   563b80e04968 i   /home/francois/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:866 (12972fa089c0 @ 1152)
```

My experience is that this problem occurs when there is ressource intensive task running (opening a game with Lutris, opening an IDE, opening an electron app) and you try to open something else.

And the problem never occured on older version of Gnome.
